rotkehlchen/chain/evm/accounting/interfaces.py

```diff
@@ -51,63 +60,194 @@ def reset(self) -> None:
+                log.debug(f'Could not consume event nr. {idx} for {event.counterparty} deposit')  # noqa: E501
                 return
 
             if next_event.balance.amount == ZERO:
                 continue
 
-            pot.add_asset_change_event(
-                method=method,
-                event_type=AccountingEventType.TRANSACTION_EVENT,
-                notes=next_event.notes if next_event.notes else '',
-                location=next_event.location,
-                timestamp=next_event.get_timestamp_in_sec(),
-                asset=next_event.asset,
-                amount=next_event.balance.amount,
-                taxable=False,  # Deposits and withdrawals are not taxable
-                count_entire_amount_spend=False,
-                count_cost_basis_pnl=False,
-                extra_data={'tx_hash': next_event.serialized_event_identifier},
+            usd_value = self._query_historical_price(pot=pot, event=next_event)
+            # balancer special case handling of refunds which are not deposits
+            if (
+                next_event.event_type == HistoryEventType.RECEIVE and
+                next_event.event_subtype == HistoryEventSubType.REMOVE_ASSET
+            ):
+                self.amounts_deposited[next_event.asset] -= Balance(
+                    amount=next_event.balance.amount,
+                    usd_value=usd_value,
+                )
+            else:
+                self.amounts_deposited[next_event.asset] += Balance(
+                    amount=next_event.balance.amount,
+                    usd_value=usd_value,
+                )
+
+            self._add_asset_change(
+                pot=pot,
+                method='spend',
+                event=next_event,
+            )
+
+    @staticmethod
+    def _query_historical_price(event: 'EvmEvent', pot: 'AccountingPot') -> FVal:
+        """Queries historical USD price for an asset."""
+        try:
+            usd_price = PriceHistorian().query_historical_price(
+                from_asset=event.asset,
+                to_asset=A_USD,
+                timestamp=ts_ms_to_sec(event.timestamp),
             )
+            usd_value = usd_price * event.balance.amount
+        except NoPriceForGivenTimestamp as e:
+            usd_value = ZERO
+            pot.cost_basis.missing_prices.add(MissingPrice(
+                from_asset=e.from_asset,
+                to_asset=e.to_asset,
+                time=e.time,
+                rate_limited=e.rate_limited,
+            ))
+
+        return usd_value
+
+    def _calculate_pnl(self, pot: 'AccountingPot', event: 'EvmEvent', usd_value: FVal) -> None:
+        """Checks if the withdrawal event causes a profit/loss.
+
+        If it does, it adds a taxable acquisition, otherwise no-op.
+        """
+        self.amounts_deposited[event.asset] -= Balance(
+            amount=event.balance.amount,
+            usd_value=usd_value,
+        )
```

Comment:
can you mention in the docstring what kind of events are going trough this logic? Is event.asset here the LP token or the underlying tokens?